### DISCUSSION TITLE
onedrive authorization hangs after logging in at login.microsoftonline.com

### DISCUSSION THREAD

**Discussion by usabong** [abraunegg/onedrive#2730] [General]
onedrive stopped connecting so I tried rebuilding, and re-installing the Linux onedrive client. When I run onedrive and it asks me to enter the authorisation url:
Configuring Global Azure AD Endpoints
Authorize this app visiting 
https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=...

After entering the login and password, it just hangs in the browser. No uri is returned. 

Onedrive is working for the same account on other devices.

**Comment by usabong** [comment]
here is the full authorisation url onedrive generated:

https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=d50ca740-c83f-4d1b-b616-12c519384f0c&scope=Files.ReadWrite%20Files.ReadWrite.All%20Sites.ReadWrite.All%20offline_access&response_type=code&prompt=login&redirect_uri=https://login.microsoftonline.com/common/oauth2/nativeclient

**Comment by abraunegg** [comment]
@usabong 
Please *re-read* the documentation very carefully:
* https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#authorize-the-application-with-your-onedrive-account
* https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-2/docs/usage.md#first-steps

Nothing has hung at all ... you are being presented with a 100% white page - you take the response URI from browser and paste it back into the client at the respective prompt.

**Reply by usabong** [reply]
I had read and re-read those links before I posted a question, but I was still expecting some sort of message returned in the web page. I did not notice that the url changed and was looking for some response on the web page, but the response is the change in the url itself.
Now I get it, and that uri worked.  Thanks for the quick response.
